SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/SpatialEventMessages.h

```diff
@@ -0,0 +1,307 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "SpatialCommonTypes.h"
+
+#include <WorkerSDK/improbable/c_worker.h>
+
+#include "SpatialEventMessages.generated.h"
+
+static_assert(sizeof(Worker_EntityId) == sizeof(int64), "EntityId assumed 64-bit here");
+static_assert(sizeof(Worker_ComponentId) == sizeof(uint32), "ComponentId assumed 32-bit here");
+static_assert(sizeof(VirtualWorkerId) == sizeof(uint32), "VirtualWorkerId assumed 32-bit here");
+static_assert(sizeof(TraceKey) == sizeof(int32), "TraceKey assumed 32-bit here");
+static_assert(sizeof(Worker_RequestId) == sizeof(int64), "RequestId assumed 64-bit here");
+
+USTRUCT()
+struct FEventMessage
+{
+	GENERATED_BODY()
+
+	FEventMessage() = default;
+	FEventMessage(const char* InType)
+		: Type(InType)
+	{}
+
+	const char* GetType() const { return Type; }
+
+private:
+
+	const char* Type = "Null";
+};
+
+USTRUCT()
+struct FEventGenericMessage : public FEventMessage
+{
+	GENERATED_BODY()
+
+	FEventGenericMessage()
```

Comment:
I think it's worth forcing using a non-default constructor for these events to ensure all calls supply the relevant information (like actors, entity ids etc)